.devcontainer/Dockerfile

```diff
@@ -1,5 +1,5 @@
 # This should be kept in sync with the python version used in docker/Dockerfile
-FROM --platform=linux/amd64 mcr.microsoft.com/devcontainers/python:3.11-bullseye@sha256:105bf6a63ab025206f019a371a735fec6553db0be520030c7a2fd0e002947232
+FROM --platform=linux/amd64 mcr.microsoft.com/devcontainers/python:3.11-bullseye@sha256:105bf6a63ab025206f019a371a735fec6553db0be520030c7a2fd0e002947232 AS devcontainer_amd64
```

Comment:
fixes:
```
WARN: FromPlatformFlagConstDisallowed: FROM --platform flag should not use constant value "linux/amd64"
```
by using stage names as described in https://docs.docker.com/reference/build-checks/from-platform-flag-const-disallowed/